Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -108,22 +108,22 @@ public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime sta
 				calList = calManager.Database.GetRecordsWithQuery(calQuery, 0, 0);
 
 				if (calList.Count == 0)
-					return Task.FromResult<IList<CalendarEvent>>(calendars);
+					return Task.FromResult<IList<CalendarEvent>>(calEvent);
 
 				calList.MoveFirst();
-				calRecord = calList.GetCurrentRecord();
 
 				do
 				{
+					calRecord = calList.GetCurrentRecord();
 					string summary = calRecord.Get<string>(Event.Summary);
-					CalendarTime startTime = calRecord.Get<CalendarTime>(Event.Start);
-					CalendarTime endTime = calRecord.Get<CalendarTime>(Event.End);
+					int IsAllday = calRecord.Get<int>(Event.IsAllday);
+					CalendarTime startTime = Convert.ToBoolean(IsAllday) ? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)) : calRecord.Get<CalendarTime>(Event.Start);
+					CalendarTime endTime = Convert.ToBoolean(IsAllday) ? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)) : calRecord.Get<CalendarTime>(Event.End);
```

Comment:
Looking more closely at this (and the `GetEventById` version), the use of `ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))` seems wrong? An all-day event may not necessarily require start/end _times_, but it certainly still requires start/end _days_...? I had previously assumed that you were just getting the date somewhere else (that maybe Tizen stored it differently), but I'm not seeing that...

The iOS/Android implementations just return whatever start/end times come from the native record. Unless there's a Tizen-specific reason not to do that, perhaps this should do the same? (i.e. just use the `CalendarTime startTime = calRecord.Get<CalendarTime>(Event.Start);` pattern, both here and in `GetEventById`)